api/comments/serializers.py

```diff
@@ -0,0 +1,146 @@
+        self._id = user_id
+        self.category = category
+        self.text = text
+
+
+class CommentSerializer(JSONAPISerializer):
+
+    filterable_fields = frozenset([
+        'deleted'
+    ])
+
+    id = IDField(source='_id', read_only=True)
+    type = TypeField()
+    content = AuthorizedCharField(source='return_content')
+
+    user = JSONAPIHyperlinkedRelatedField(view_name='users:user-detail', lookup_field='pk', lookup_url_kwarg='user_id', link_type='related', read_only=True)
+    node = JSONAPIHyperlinkedRelatedField(view_name='nodes:node-detail', lookup_field='pk', lookup_url_kwarg='node_id', link_type='related', read_only=True)
+    target = JSONAPIHyperlinkedGuidRelatedField(link_type='related', meta={'type': 'get_target_type'})
+    replies = JSONAPIHyperlinkedIdentityField(view_name='comments:comment-replies', lookup_field='pk', link_type='self', lookup_url_kwarg='comment_id')
+    reports = JSONAPIHyperlinkedIdentityField(view_name='comments:comment-reports', lookup_field='pk', lookup_url_kwarg='comment_id', link_type='related', read_only=True)
+
+    date_created = ser.DateTimeField(read_only=True)
+    date_modified = ser.DateTimeField(read_only=True)
+    modified = ser.BooleanField(read_only=True, default=False)
+    deleted = ser.BooleanField(read_only=True, source='is_deleted', default=False)
+
+    # LinksField.to_representation adds link to "self"
+    links = LinksField({})
+
+    class Meta:
+        type_ = 'comments'
+
+    def create(self, validated_data):
+        user = validated_data['user']
+        node = validated_data['node']
+
+        validated_data['content'] = validated_data.pop('return_content')
+        if node and node.can_comment(Auth(user)):
+            comment = Comment.create(auth=Auth(user), **validated_data)
+        else:
+            raise PermissionDenied("Not authorized to comment on this project.")
+        return comment
+
+    def update(self, comment, validated_data):
+        assert isinstance(comment, Comment), 'comment must be a Comment'
+        auth = Auth(self.context['request'].user)
+        if validated_data:
+            if 'return_content' in validated_data:
+                comment.edit(validated_data['return_content'], auth=auth, save=True)
+            is_deleted = validated_data.get('is_deleted', None)
+            if is_deleted:
+                comment.delete(auth, save=True)
+            elif comment.is_deleted:
+                comment.undelete(auth, save=True)
+        return comment
+
+    def get_target_type(self, obj):
+        target_id = obj._id
+        guid_object = Guid.load(target_id)
+        if not guid_object:
+            raise NotFound('Comment target not found.')
+        target = guid_object.referent
+
+        if isinstance(target, Node):
+            return 'node'
+        elif isinstance(target, Comment):
+            return 'comment'
+        else:
+            raise InvalidModelValueError('Invalid comment target.')
```

Comment:
There appears to be a `_name` on all GuidStoredObjects. On comments it's 'comment' and on nodes it's 'node'. I bet you could get away with:
```python
    def get_target_type(self, obj):
        return obj._name
```